index.js

```diff
@@ -2320,11 +2317,35 @@
 
   //. ### List
   //.
-  //. The List type represents non-Function values with integer `length`
-  //. properties greater than or equal to zero, such as `[1, 2, 3]` and
-  //. `'foo'`.
+  //. The List type constructor enables type signatures to describe ad hoc
+  //. polymorphic functions which operate on either [`Array`][$.Array] or
+  //. [`String`][$.String] values.
   //.
-  //. `[a]` is the notation used to represent a List of values of type `a`.
+  //. Mental gymnastics are required to treat arrays and strings similarly.
+  //. `[1, 2, 3]` is a list containing `1`, `2`, and `3`. `'abc'` is a list
+  //. containing `'a'`, `'b'`, and `'c'`. But what is the type of `'a'`?
+  //. `String`, since JavaScript has no Char type! So:
+  //.
+  //.     'abc' :: String, List String, List (List String), ...
+  //.
+  //. Every member of `String` is also a member of `List String`! This
+  //. affects the interpretation of type signatures. Consider the type of
+  //. [`indexOf`](#indexOf):
+  //.
+  //.     a -> List a -> Maybe Integer
+  //.
+  //. Assume the second argument is `'hello' :: List String`. `a` must then be
+  //. replaced with `String`:
+  //.
+  //.     String -> List String -> Maybe Integer
+  //.
+  //. Since `List String` and `String` are interchangeable, the former can be
+  //. replaced with the latter:
+  //.
+  //.     String -> String -> Maybe Integer
+  //.
+  //. It's then apparent that the first argument needn't be a single-character
+  //. string; the correspondence between arrays and strings does not hold.
```

Comment:
What if we we also add `Char` type which checks if length of a `String` is 1?
Then we can say that `String` is `List Char`. This will have an issue with emojis `"⏰".length == 2`  but I think we can check it like this:
```js
const isChar = s => s.charAt(1) == "" && s.length != 0
isChar("⏰🗒") // false
isChar("⏰") // true
isChar("") // false
```